tools/crm_attribute.c

```diff
@@ -503,9 +503,20 @@ main(int argc, char **argv)
         if (target != NULL) {
             g_free(options.dest_uname);
             options.dest_uname = g_strdup(target);
+        } else if (getenv("CIB_file") != NULL) {
+            char *hostname = pcmk_hostname();
+
+            g_free(options.dest_uname);
+
+            /* This silliness is so that dest_uname is always a glib-managed
+             * string so we know how to free it later.  pcmk_hostname returns
+             * a newly allocated string via strdup.
+             */
+            options.dest_uname = g_strdup(hostname);
+            free(hostname);
         }
 
-        if (options.dest_uname == NULL) {
+        if (options.dest_uname == NULL && getenv("CIB_file") == NULL) {
```

Comment:
Shouldn't be necessary, if dest_uname is null then CIB_file definitely is too at this point

Reply:
Yes, I think you are right.  Fixed.